### PR TITLE
[ASCII-1020] Convert comp/core/status and comp/core/status/statusimpl into modules

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

I hit a blocker while [adding](https://github.com/DataDog/datadog-agent/pull/22115) the status component to the agent. The blocker is that the `defaultforwarder` component was recently converted to a Go module on the PR that uses the new status component we add the `comp/core/status` as new dependency into the `defaultforwarder` component ([commit](https://github.com/DataDog/datadog-agent/pull/22115/commits/453cda1610b8e2f00a80f8a0e2e93b1fe6696436#diff-a9021d2048835e7c123eb407a002586b10adc1101ff02568bf5e3ef4158e8affR13))

The solution is to create two separate modules for the status component.

### Minor changes 
The `"github.com/DataDog/datadog-agent/pkg/collector/python"` and `"github.com/DataDog/datadog-agent/pkg/util/flavor"` are not modules, so I can not use them directly in the new status module. I decided to inject the values I was extracting from those packages as a new dependency as part of the status component 

### Describe how to test/QA your changes

- Being able to import the `comp/core/status` and `comp/core/status/statusimpl` properly in a separate project

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/core/status/statusimpl/go.mod [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	github.com/DataDog/datadog-agent/comp/core/status => ../.
+	github.com/DataDog/datadog-agent/pkg/config/setup => ../../../../pkg/config/setup
+	github.com/DataDog/datadog-agent/pkg/util/fxutil => ../../../../pkg/util/fxutil
+	github.com/DataDog/datadog-agent/pkg/version => ../../../../pkg/version
+)
+
+require (
+	github.com/DataDog/datadog-agent/comp/core/config v0.0.0-00010101000000-000000000000
+	github.com/DataDog/datadog-agent/comp/core/status v0.0.0-00010101000000-000000000000
+	github.com/DataDog/datadog-agent/pkg/config/setup v0.51.0-rc.2
```
Comment (dineshg13) [inline comp/core/status/statusimpl/go.mod:16]: Whats the need to create both the modules ?

File: pkg/cli/subcommands/check/command.go [hunk 2]
```diff
@@ -178,6 +178,7 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 				}),
 
 				fx.Supply(
+					status.Params{},
```
Comment (GustavoCaso) [inline pkg/cli/subcommands/check/command.go:181]: For the check commands we are ok just with empty params

File: comp/core/status/statusimpl/go.mod [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	github.com/DataDog/datadog-agent/pkg/config/env v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/config/model v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/executable v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/filesystem v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/hostname/validate v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/log v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/optional v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/scrubber v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/system/socket v0.51.0-rc.2 // indirect
+	github.com/DataDog/datadog-agent/pkg/util/winutil v0.51.0-rc.2 // indirect
```
Comment (dineshg13) [inline comp/core/status/statusimpl/go.mod:36]: Can you please add replace all the modules from Agent repo.

File: go.mod [hunk 4]
```diff
@@ -242,7 +244,7 @@ require (
 	go.uber.org/atomic v1.11.0
 	go.uber.org/automaxprocs v1.5.3
 	go.uber.org/dig v1.17.0
-	go.uber.org/fx v1.18.2
+	go.uber.org/fx v1.20.1
```
Comment (dineshg13) [inline go.mod:247]: Don't update fx. Make sure to set fx version  `v1.18.2` in all modules. There are some known issued with the upgrade

**Review by dineshg13 — APPROVED** [review]
LGTM

File: comp/core/status/statusimpl/common_header_provider.go [hunk 5]
```diff
@@ -68,24 +67,24 @@ func (h *headerProvider) HTML(buffer io.Writer) error {
 	return t.Execute(buffer, h.data)
 }
 
-func newCommonHeaderProvider(config config.Component) status.HeaderProvider {
+func newCommonHeaderProvider(params status.Params, config config.Component) status.HeaderProvider {
 
 	data := map[string]interface{}{}
 	data["version"] = version.AgentVersion
-	data["flavor"] = flavor.GetFlavor()
```
Comment (hush-hush) [inline comp/core/status/statusimpl/common_header_provider.go:75]: Is `GetFlavorr` and `GetHumanReadableFlavor` the same ?

File: comp/core/status/component.go [hunk 6]
```diff
@@ -25,6 +25,12 @@ type Component interface {
 	GetStatusBySection(section string, format string, verbose bool) ([]byte, error)
 }
 
+// Params store configurable options for the status component
+type Params struct {
+	Flavor        string
```
Comment (hush-hush) [inline comp/core/status/component.go:30]: Flavor being such a small package, would it not be easier to convert it as a module to rather that take is as a parameter ?

File: comp/core/status/statusimpl/common_header_provider.go (hunk 5 above)
Comment (GustavoCaso) [reply comp/core/status/statusimpl/common_header_provider.go:75]: It is not the same. I was thinking of cutting corners and only use `GetHumanReadableFlavor` when populating the params

File: comp/core/status/component.go (hunk 6 above)
Comment (GustavoCaso) [reply comp/core/status/component.go]: Done https://github.com/DataDog/datadog-agent/pull/22153/commits/02f45bc9441521dc1f01b474e53e6fbd63748588 and https://github.com/DataDog/datadog-agent/pull/22153/commits/57790ac17fe42f6e75f0f36041322459028a169a

**Review by hush-hush — APPROVED** [review]

**Review by chouquette — APPROVED** [review]
looks good for agent-platform owned file!

**Pull request merged**